src/content/speakers.astro

```diff
@@ -1,13 +1,71 @@
 ---
-import type { Person } from "../data/types";
+import type {
+  PersonalDetails,
+  Speaker,
+  Support,
+  Ambassador,
+  TalkDetails,
+} from "../data/types";
 import { CFP_OPEN } from "../data/constants";
 
 import PersonCard from "../components/PersonCard.astro";
 import PlaceholderCard from "../components/PlaceholderCard.astro";
+import { getPersonIdFromFilePath } from "../utils";
+import type { MarkdownInstance } from "astro";
 
-const speakers = await Astro.glob<Person>("../data/speakers/*.md");
-const ambassadors = await Astro.glob<Person>("../data/ambassadors/*.md");
-const support = await Astro.glob<Person>("../data/support/*.md");
+const speakers = await Astro.glob<PersonalDetails>("../data/speakers/*.md");
+const speakerTalks = await Astro.glob<TalkDetails>(
+  "../data/speaker-talks/*.md"
+);
+const ambassadors = await Astro.glob<PersonalDetails>(
+  "../data/ambassadors/*.md"
+);
+const support = await Astro.glob<PersonalDetails>("../data/support/*.md");
+
+const getSpeakerInfo = (
+  speakerMd: MarkdownInstance<PersonalDetails>
+): Speaker => {
+  const speakerId = getPersonIdFromFilePath(speakerMd.file);
+
+  const speakerTalk = speakerTalks.find(
+    (s) => speakerId === getPersonIdFromFilePath(s.file)
+  );
+
+  return {
+    type: "speaker",
+    id: speakerId,
+    bio: speakerMd.frontmatter,
+    BioMdContent: speakerMd.Content,
+    talk: speakerTalk ? speakerTalk.frontmatter : null,
+    TalkMdContent: speakerTalk ? speakerTalk.Content : null,
```

Comment:
```suggestion
    talk: speakerTalk?.frontmatter,
    TalkMdContent: speakerTalk?.Content,
```

Reply:
This is a similar example with the comment above:
 - null marks a value that is not there and won't be there in the future
 - undefined can mean an error in data, or (not the case here) async data that hasn't been defined yet